File: primeSum.py
import math

def isPrime(n):

    if (n < 1):
        return False

    for i in range(2, int(math.sqrt(n) + 1)):
        if (n % i == 0):
            return False

    return True

def primesum(A):
    primes = [True] * (A + 1)

    primes[0] = False
    primes[1] = False

    sqrt_n = int(math.sqrt(A))

    for i in range(2, sqrt_n + 1):
        if (primes[i]):
            for j in range(i * 2, A + 1, i):
                primes[j] = False
    
    result = [i for i, b in enumerate(primes) if b]

    print(result)
        
    # Basically there are always two prime numbers that sum up to a given number
    # After generating the list of primes we loop through it and check each's complement
    # if the complement of the current prime number is also a prime, then it sums up to the number
    for i in range(len(result)):
        if (isPrime(A - result[i])):
            return sorted([result[i], A - result[i]])
    

print(primesum(4))

File: primeSum.js


const primeSum = (n) => {
    let primes = [];
    for  (let i = 0; i <= n; i++) {
        primes[i] = true;
    }

    // 0 and 1 are not primes
    primes[0] = false;
    primes[1] = false;

    // sieve of eratosthenes
    for (let i = 2; i <= Math.floor(Math.sqrt(n)); i++) {
        if (primes[i]) {
            for (let j = i * 2; j <= n; j+=i) {
                primes[j] = false;
            }
        }
    }
    const primeNumbers = [];

    // get the prime numbers
    for (let i = 0;  i < primes.length; i++) {
        if (primes[i]) {
            primeNumbers.push(i)
        }
    }

    console.log(primeNumbers);

    // test for primeness
    const isPrime = (num) => {
        for (let i = 2; i <= Math.floor(Math.sqrt(num)); i++) {
            if (num % i === 0) {
                return false;
            }
        }
        return true;
    }

    // if the complement of the prime number is also prime, then we have the solution
    for (let i = 0; i < primeNumbers.length; i++) {
        if (isPrime(n - primeNumbers[i])) {
            return [primeNumbers[i], n - primeNumbers[i]]
        }
    }


    return primeNumbers;
}

console.log(primeSum(11))